Allow AVX builds for Windows with clang.

The check exists because GCC does not realign the stack for 32-byte spills on
Windows. Clang does.

--- a/src/util/util_bit.h
+++ b/src/util/util_bit.h
@@ -12,7 +12,7 @@
 
 #ifdef DXVK_ARCH_X86
   #ifndef _MSC_VER
-    #if defined(_WIN32) && (defined(__AVX__) || defined(__AVX2__))
+    #if !defined(__clang__) && defined(_WIN32) && (defined(__AVX__) || defined(__AVX2__))
       #error "AVX-enabled builds not supported due to stack alignment issues."
     #endif
     #if defined(__WINE__) && defined(__clang__)
